config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java

```diff
@@ -1907,6 +1957,234 @@ public AuthorizeExchangeSpec access(ReactiveAuthorizationManager<AuthorizationCo
+			return this;
+		}
+
+		public SessionManagementSpec sessionAuthenticationStrategy(
+				ServerSessionAuthenticationStrategy sessionAuthenticationStrategy) {
+			Assert.notNull(sessionAuthenticationStrategy, "sessionAuthenticationStrategy cannot be null");
+			this.providedSessionAuthenticationStrategy = sessionAuthenticationStrategy;
+			return this;
+		}
+
+		void configure(ServerHttpSecurity http) {
+			if (this.concurrentSession != null) {
+				ReactiveSessionRegistry reactiveSessionRegistry = getSessionRegistry();
+				ConcurrentSessionControlServerSessionAuthenticationStrategy concurrentSessionControlStrategy = new ConcurrentSessionControlServerSessionAuthenticationStrategy(
+						reactiveSessionRegistry);
+				concurrentSessionControlStrategy.setMaximumSessions(this.maxSessions);
+				concurrentSessionControlStrategy.setMaximumSessionsExceededHandler(this.maximumSessionsExceededHandler);
+				RegisterSessionServerSessionAuthenticationStrategy registerSessionAuthenticationStrategy = new RegisterSessionServerSessionAuthenticationStrategy(
+						reactiveSessionRegistry);
+				this.sessionAuthenticationStrategy = new DelegatingServerSessionAuthenticationStrategy(
+						concurrentSessionControlStrategy, this.providedSessionAuthenticationStrategy,
+						registerSessionAuthenticationStrategy);
+				SessionRegistryWebFilter sessionRegistryWebFilter = new SessionRegistryWebFilter(
+						reactiveSessionRegistry);
+				http.addFilterAfter(sessionRegistryWebFilter, SecurityWebFiltersOrder.HTTP_HEADERS_WRITER);
+			}
+			else {
+				this.sessionAuthenticationStrategy = this.providedSessionAuthenticationStrategy;
+			}
+		}
+
+		private ReactiveSessionRegistry getSessionRegistry() {
+			if (this.sessionRegistry == null) {
+				this.sessionRegistry = getBeanOrNull(ReactiveSessionRegistry.class);
+			}
+			if (this.sessionRegistry == null) {
+				throw new IllegalStateException(
+						"A ReactiveSessionRegistry needs to be specified to enable concurrent session management");
+			}
+			return this.sessionRegistry;
+		}
+
+		/**
+		 * Configures how many sessions are allowed for a given user.
+		 */
+		public class ConcurrentSessionsSpec {
+
+			/**
+			 * Sets the {@link ReactiveSessionRegistry} to use.
+			 * @param reactiveSessionRegistry the {@link ReactiveSessionRegistry} to use
+			 * @return the {@link ConcurrentSessionsSpec} to continue customizing
+			 */
+			public ConcurrentSessionsSpec sessionRegistry(ReactiveSessionRegistry reactiveSessionRegistry) {
+				SessionManagementSpec.this.sessionRegistry = reactiveSessionRegistry;
+				return this;
+			}
+
+			/**
+			 * Sets the {@link Function} to use to determine the maximum number of
+			 * sessions for a user.
+			 * @param maxSessions the {@link Function} to use
+			 * @return the {@link ConcurrentSessionsSpec} to continue customizing
+			 */
+			public ConcurrentSessionsSpec maxSessions(Function<Authentication, Mono<Integer>> maxSessions) {
```

Comment:
Having a `Function` as an API can be a bit more challenging. Have you considered having two methods:

```java
.maxSessions(Function)

.maxSessions(Integer)
```

In this way, folks can supply a global max, should that be preferred, in a way that is clearly correct.